sampler_options.go

```diff
@@ -32,6 +34,7 @@ type samplerOptions struct {
 	sampler       Sampler
 	logger        Logger
 	hostPort      string
+	timer         *time.Ticker
```

Comment:
I am not sure about this. Why not just store the parameter, and only deal with timer in one place?